gc/src/lib.rs

```diff
@@ -30,17 +32,13 @@ pub use gc::force_collect;
 ///
 /// See the [module level documentation](./) for more details.
 pub struct Gc<T: Trace + ?Sized + 'static> {
-    // XXX We can probably take advantage of alignment to store this
-    root: Cell<bool>,
-    _ptr: NonZero<*mut GcBox<T>>,
+    _ptr: UnsafeCell<NonZero<*mut GcBox<T>>>,
 }
 
 impl<T: ?Sized> !marker::Send for Gc<T> {}
 
 impl<T: ?Sized> !marker::Sync for Gc<T> {}
 
-impl<T: Trace + ?Sized + marker::Unsize<U>, U: Trace + ?Sized> CoerceUnsized<Gc<U>> for Gc<T> {}
```

Comment:
I believe that the CoerceUnsized constraints are a bit too strong in rustc. I believe NonZero allows for the CoerceUnsize because of this: https://github.com/rust-lang/rust/blob/f9b8c49cdb75fb571c7b3ea4af90b0e96929276c/src/libcore/nonzero.rs#L78. UnsafeCell doesn't have such an impl for it (although I believe it would be safe for it to have one).